lib/rucio/daemons/rsedecommissioner/profiles/generic.py

```diff
@@ -0,0 +1,512 @@
+            # -> Update the state to UNAVAILABLE
+            if state == ReplicaState.COPYING:
+                # Should we allow requests in certain states? Depends on the
+                # details of the request state machine
+                count_stmt = select(
+                    func.count()
+                ).select_from(
+                    select(
+                        models.Request.id
+                    ).filter(
+                        models.Request.scope == scope,
+                        models.Request.name == name,
+                        models.Request.dest_rse_id == rse['id']
+                    ).subquery()
+                )
+
+                num_requests = session.execute(count_stmt).scalar_one()
+
+                if num_requests == 0:
+                    logger(logging.INFO,
+                           '(%s) Replica %s:%s is in COPYING state without a'
+                           ' valid request. Changing state to UNAVAILABLE.',
+                           rse['rse'], scope, name)
+                        
+                    replica.state = ReplicaState.UNAVAILABLE
+                    updated = True
+                    
+            reap_pending = False
+
+        # 2. Condition on the lock count
+        if lock_cnt != 0:
+            # 2.1 No actual lock -> Reset the lock count
+            count_stmt = select(
+                func.count()
+            ).select_from(
+                select(
+                    models.ReplicaLock.rule_id
+                ).filter(
+                    models.ReplicaLock.scope == scope,
+                    models.ReplicaLock.name == name,
+                    models.ReplicaLock.rse_id == rse['id']
+                ).subquery()
+            )
+
+            actual_lock_count = session.execute(count_stmt).scalar_one()
+            
+            if actual_lock_count == 0:
+                logger(logging.INFO,
+                       '(%s) Replica %s:%s has lock count %s but zero actual'
+                       ' locks. Changing lock_cnt to 0.',
+                       rse['rse'], scope, name, lock_cnt)
+                    
+                replica.lock_cnt = 0
+                updated = True
+                
+            reap_pending = False
+        
+        # 3. Tombstone missing or in the future -> RIP now
+        if tombstone is None or tombstone >= datetime.utcnow():
+            logger(logging.INFO, 'Marking tombstone of replica %s:%s as UTCNOW.',
+                   scope, name)
+                
+            replica.tombstone = datetime.utcnow()
+            updated = True
+            
+            reap_pending = False
+            
+        if updated:
+            num_updated += 1
+           
+        if reap_pending:
+            num_waiting_reap += 1
+            
+    session.flush()
+    
+    logger(logging.INFO, '(%s) %s replicas have been updated. %s replicas should'
+           ' be reaped soon. %s replicas remain unhandled.',
```

Comment:
I would avoid writing ‘soon’. It could be that the operators have neglected to assign a Reaper instance to this RSE. It’s better to phrase it as ‘pending deletion’.